sdk/workflow-tengo/src/slices.lib.tengo

```diff
@@ -140,18 +140,27 @@ map := func(slice, fn) {
  * @return final result: any - the final result
  */
 reduce := func(slice, fn, ...init) {
-	result := undefined
+	result := len(init) > 0 ? init[0] : undefined
 	for i, elem in slice {
-		if i == 0 {
-			result = len(init) > 0 ? init : elem
-		} else {
-			result = fn(result, elem)
-		}
+		result = fn(result, elem)
 	}
-
 	return result
 }
```

Comment:
👍 
Have you checked if it is used anywhere? Maybe we should change `init` from optional to required.